streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -294,11 +306,13 @@ public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
         if (state == KafkaStreams.State.CREATED) {
-            for (final StreamThread thread : threads)
-                thread.start();
-
             setState(KafkaStreams.State.RUNNING);
 
+            for (final StreamThread thread : threads) {
+                threadState.put(thread.getId(), thread.state());
```

Comment:
Good catch, thanks. Ideally I'd prefer just one point of updating and enforcing that. I'll pull out this code and place it slightly higher.